test/utils/runners.go

```diff
@@ -52,6 +52,38 @@ const (
 	nonExist = "NonExist"
 )
 
+func RunPodAndGetNodeName(c clientset.Interface, pod *v1.Pod, timeout time.Duration) (string, error) {
+	retries := 5
+	name := pod.Name
+	namespace := pod.Namespace
+	var err error
+	// Create a Pod
+	for i := 0; i < retries; i++ {
+		_, err = c.Core().Pods(namespace).Create(pod)
+		if err == nil || apierrs.IsAlreadyExists(err) {
+			break
+		}
+	}
+	if err != nil && !apierrs.IsAlreadyExists(err) {
+		return "", err
+	}
+	// Wait until it's scheduled
```

Comment:
Waiting until scheduled seems like a generic enough functionality to be separated into separate function.